buzsaki_lab_to_nwb/tingley_code_septal/tingleyseptalbehaviorinterface.py

```diff
@@ -0,0 +1,131 @@
+"""Authors: Heberto Mayorquin and Cody Baker."""
+from pathlib import Path
+import warnings
+
+import numpy as np
+from hdmf.backends.hdf5.h5_utils import H5DataIO
+
+from pynwb.file import NWBFile, TimeIntervals
+from pynwb.behavior import SpatialSeries, Position
+from nwb_conversion_tools.basedatainterface import BaseDataInterface
+from nwb_conversion_tools.utils.conversion_tools import get_module
+from nwb_conversion_tools.utils.json_schema import FolderPathType
+
+from .tingleyseptal_utils import read_matlab_file
+
+
+class TingleySeptalBehaviorInterface(BaseDataInterface):
+    """Behavior data interface for the Tingley Septal project."""
+
+    def __init__(self, folder_path: FolderPathType):
+        super().__init__(folder_path=folder_path)
+
+    def run_conversion(self, nwbfile: NWBFile, metadata: dict):
+        session_path = Path(self.source_data["folder_path"])
+        session_id = session_path.stem
+
+        # Position
+        module_name = "Position"
+        module_description = "Contains behavioral data concerning position."
+        processing_module = get_module(nwbfile=nwbfile, name=module_name, description=module_description)
+
+        behavior_file_path = Path(session_path) / f"{session_id}.behavior.mat"
+
+        behavior_mat = read_matlab_file(str(behavior_file_path))["behavior"]
+        time_stamps = behavior_mat["timestamps"]
+        starting_time = time_stamps[0][0]
+        rate = behavior_mat["samplingRate"]
+
+        position = behavior_mat["position"]
+        pos_data = [[x, y, z] for (x, y, z) in zip(position["x"], position["y"], position["y"])]
+        pos_data = np.array(pos_data)[:, :, 0]
+
+        try:
+            units = behavior_mat["units"]
+            if units == "m":
+                conversion = 1.0
+            else:
+                warnings.warn(f"Spatial units {units} not listed in meters; " "setting conversion to nan.")
+                conversion = np.nan
+        except KeyError:
+            warnings.warn(f"Units not available")
+            units = None
+            conversion = np.nan
+        try:
+            description = behavior_mat["description"].replace("/", "")
+        except KeyError:
+            description = "generic_position_tracking"
```

Comment:
```suggestion
            description = behavior_mat.get("description", "").replace("/", "")
```
`Try`/`except` catching is really only for situations where you can't do an `if` check or some other safe operation. It shouldn't be a general habit, even with non-bare `except`.

Can you change the rest of these similarly?

Reply:
The second argument of `dictionary_object.get("key_name", "default")` is what is returned if the `key_name` is not present.

Reply:
Great. I did not know this trick with dictionaries. I did feel uncomfortable with the try-excepts block and I wanted to ask you if there a way around it. I will change.